### ISSUE TITLE
Application hang (or focus) issue after clearing private data and hitting back

### ISSUE THREAD

**Issue by AaronMT** [mozilla-mobile/firefox-tv#1592]
* Build: 3.1.3
* 4K Device 

Launch 3.1.3

- Settings, Clear Private Data
- Hit `back` on the device remote

Application either hangs or loses focus and is thus unusable from that point onwards.

**Comment by AaronMT** [comment]
This seems reproducible in 3.1.2 too.

**Comment by Baron-Severin** [comment]
@AaronMT I'm unable to reproduce on v3.1.2, could you verify?  I'm testing on a Cube, I'll double check on other devices.

AFAICT, this was introduced in b81db7f6ba24fd2dad2c32c2c1c6ca15dc869734.

**Comment by Baron-Severin** [comment]
Just to document some weirdness...

- Open settings
- Clear private data
- Hit back on remote remote
- Hit menu on remote twice
(Turbo mode button appears to be focused)
- Hit any direction keys on the d-pad
(Nothing will happen)
- Click
(Pocket will appear to be clicked, app will crash with the following KNPE)

```
2018-12-19 09:57:21.117 24159-24159/org.mozilla.tv.firefox.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.tv.firefox.debug, PID: 24159
    kotlin.KotlinNullPointerException
        at org.mozilla.tv.firefox.webrender.VideoVoiceCommandMediaSession.onDestroyWebView(VideoVoiceCommandMediaSession.kt:132)
        at org.mozilla.tv.firefox.webrender.WebRenderFragment.onDestroyView(WebRenderFragment.kt:320)
        at android.support.v4.app.Fragment.performDestroyView(Fragment.java:2678)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1533)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:797)
        at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2625)
        at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
        at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2366)
        at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2273)
        at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:733)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6188)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:911)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:801)
```

**Comment by Baron-Severin** [comment]
More clues.

The implementation of this that didn't cause any focus issues hit this code when backing after clearing data.

```
context!!.webRenderComponents.sessionManager.remove()
if (!browserOverlay.isVisible) {
    setOverlayVisible(true)
} else {
    return false
}
```

Interestingly, even though the overlay is clearly visible to the user, `!browserOverlay.isVisible` is evaluating to true, so the top branch is hit.  

TL;DR, it looks like both the old and the new code **should** return false from this use case, but the old code didn't.

**Comment by Baron-Severin** [comment]
Per @AaronMT the focus bug is also reproducible on some devices.

This is not a new regression, and as long as it is not reproducible on master we will not be fixing it on v3.1.3.

**Comment by liuche** [comment]
Actually now that the SettingsFragment crashes on data clear in #1611, this isn't even a problem anymore.
